autobot-backend/modern_ai_integration_test.py

```
@@ -0,0 +1,123 @@
+string ``{analysis_goal}``, and nothing downstream could tell: the vision call
+succeeded, the JSON parsed, and the caller got a plausible description of the
+screen that simply ignored what they asked about.
+
+The assertions here read the prompt that reached the fake model rather than the
+template that produced it. Asserting a prompt was built, or that the template
+no longer holds a replacement field, is what the defect itself would pass.
+"""
+
+import json
+from types import SimpleNamespace
+from typing import Any, List
```

Ruff (0.16.3)

[warning] `typing.List` is deprecated, use `list` instead

(UP035)